feat: support multiple versions of the plpgsql_check extension

jfroche opened the pull request:
Build multiple versions of the plpgsql_check extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

jfroche committed:
feat: support multiple versions of the plpgsql_check extension

Build multiple versions of the plpgsql_check extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

hunleyd reviewed (dismissed):
rebase to fix conflicts?

samrose committed:
tests: refactor to use the default nixos tests

jfroche committed:
fix(plpgsql-check): add missing upgrade path files between versions

Generate empty SQL files for version-to-version upgrade paths to satisfy
PostgreSQL extension upgrade requirements.

samrose committed:
feat: requires switch version script due to shared_preload

jfroche committed:
fix: test with plpgsql_check extension
